### PR TITLE
feat(cart): add update bundle item mutation and service method

### PR DESCRIPTION
**Pull request by KarolNowakk**

### CODE REVIEW

File: cart/application/cartService.go [hunk 1]
```diff
@@ -410,6 +414,68 @@ func (cs *CartService) UpdateItems(ctx context.Context, session *web.Session, up
...
+
+	if product.Type() != productDomain.TypeBundleWithActiveChoices {
+		return ErrProductNotTypeBundle
+	}
+
+	if cs.itemValidator != nil {
+		decoratedCart, _ := cs.cartReceiverService.DecorateCart(ctx, cart)
+		delivery, err := cart.GetDeliveryByItemID(updateCommand.ItemID)
+		if err != nil {
+			return fmt.Errorf("delivery code not found by item, while updateing bundle %w", err)
```
Comment (ChristopherSchwab) [inline cart/application/cartService.go:453]:
```suggestion
			return fmt.Errorf("delivery code not found by item while updating bundle %w", err)
```

File: cart/application/cartService.go [hunk 2]
```diff
@@ -410,6 +414,68 @@ func (cs *CartService) UpdateItems(ctx context.Context, session *web.Session, up
...
+		}
+
+		if err := cs.itemValidator.Validate(ctx, session, decoratedCart, delivery.DeliveryInfo.Code, cartDomain.AddRequest{}, product); err != nil {
+			return fmt.Errorf("error vaidating bundle update: %w", err)
```
Comment (ChristopherSchwab) [inline cart/application/cartService.go:457]:
```suggestion
			return fmt.Errorf("error validating bundle update: %w", err)
```

File: cart/application/cartService.go [hunk 3]
```diff
@@ -410,6 +414,68 @@ func (cs *CartService) UpdateItems(ctx context.Context, session *web.Session, up
...
+	}
+
+	product, err := cs.productService.Get(ctx, item.MarketplaceCode)
+	if err != nil {
+		cs.logger.WithContext(ctx).WithField(flamingo.LogKeySubCategory, "UpdateItemBundleConfig").Error(err)
+
+		return err
+	}
+
+	product, err = cs.getSpecificProductType(ctx, product, item.VariantMarketPlaceCode, item.BundleConfig)
```
Comment (ChristopherSchwab) [inline cart/application/cartService.go:440]: Should be updateCommand.BundleConfiguration, shouldnt it?

File: cart/application/cartService.go [hunk 4]
```diff
@@ -410,6 +414,80 @@ func (cs *CartService) UpdateItems(ctx context.Context, session *web.Session, up
...
+	product, err = cs.getBundleWithActiveChoices(ctx, product, item.VariantMarketPlaceCode, updateCommand.BundleConfiguration)
+	if err != nil {
+		return fmt.Errorf("error converting product to bundle: %w", err)
+	}
+
+	if cs.itemValidator != nil {
+		decoratedCart, _ := cs.cartReceiverService.DecorateCart(ctx, cart)
+		delivery, err := cart.GetDeliveryByItemID(updateCommand.ItemID)
+		if err != nil {
+			return fmt.Errorf("delivery code not found by item, while updating bundle %w", err)
```
Comment (carstendietrich) [inline cart/application/cartService.go:449]:
```suggestion
			return fmt.Errorf("delivery code not found by item, while updating bundle: %w", err)
```

File: cart/application/cartService.go [hunk 5]
```diff
@@ -90,7 +90,11 @@ func (e *RestrictionError) Error() string {
 	return e.message
 }
 
-var _ Service = &CartService{}
+var (
+	_                          Service = &CartService{}
+	ErrBundleConfigNotProvided         = errors.New("error no bundle config provided")
+	ErrProductNotTypeBundle            = errors.New("product not of type bundle with active choices")
```
Comment (carstendietrich) [inline cart/application/cartService.go:96]:
```suggestion
	ErrProductNotTypeBundle            = errors.New("product not of type bundle")
```

File: cart/application/cartService.go [hunk 6]
```diff
@@ -410,6 +414,80 @@ func (cs *CartService) UpdateItems(ctx context.Context, session *web.Session, up
...
+	}
+
+	product, err := cs.productService.Get(ctx, item.MarketplaceCode)
+	if err != nil {
+		cs.logger.WithContext(ctx).WithField(flamingo.LogKeySubCategory, "UpdateItemBundleConfig").Error(err)
+
+		return err
+	}
+
+	product, err = cs.getBundleWithActiveChoices(ctx, product, item.VariantMarketPlaceCode, updateCommand.BundleConfiguration)
```
Comment (carstendietrich) [inline cart/application/cartService.go:440]: guess variant marketplace code makes only sense for configurables

File: cart/interfaces/graphql/mutationresolvers.go [hunk 7]
```diff
@@ -103,6 +103,31 @@ func (r *CommerceCartMutationResolver) CommerceUpdateItemQty(ctx context.Context
 	return r.q.CommerceCart(ctx)
 }
 
+// CommerceUpdateItemBundleConfig mutation for updating item quantity
+func (r *CommerceCartMutationResolver) CommerceUpdateItemBundleConfig(ctx context.Context, itemID string, bundleConfig []*dto.ChoiceConfiguration) (*dto.DecoratedCart, error) {
```
Comment (carstendietrich) [inline cart/interfaces/graphql/mutationresolvers.go:107]: integration test where are you?

**Review by carstendietrich — CHANGES_REQUESTED** [review]

**Review by carstendietrich — APPROVED** [review]

**Pull request merged**